### PR TITLE
misc: use hash suffix instead of prefix in get_model_key

### LINKED ISSUE #213: [feature] easier way to override `get_model_key()`
Hi,
the function [utils.get_model_key](https://github.com/0b01001001/spectree/blob/9147598a03d12223fe3cdf1523d08d2390475e9c/spectree/utils.py#L220) is used for generating model IDs + their references in `openapi.json`. We try to use this as a source for generating matching models on a frontend (via `@rtk-query/codegen-openapi` ), but the model IDs are prefixed with sha1 hash (which can start with a number), and the mentioned tool cant handle that. 

I overrided `SpecTree._generate_spec()` where I manually inject a prefix everywhere, but that's quite inelegant, and the generated model names are unpleasant. 

Easy way out of this would be having `get_model_key()` defined in a way which could be overriden easily. Not sure how exactly that would be done, so I'm opening this issue for now to hear some feedback if I'm not missing something obvious here.

Thanks.

### PR DESCRIPTION
**Pull request by yedpodtrzitko**
closes #213

### CODE REVIEW

**Comment by yedpodtrzitko** [comment]
ok I actually ran the testsuite this time, since it's faster than waiting for the pipeline approval. I noticed one thing though - `get_model_path_key()` (which was making the tests fail) is being used only in the testsuite and never in the actual production code. As such, shouldnt be its definition moved into the testusite as well? Or maybe replaced with usage of `get_model_key()`.

**Comment by kemingy** [comment]
> ok I actually ran the testsuite this time before pushing. I noticed one thing though - `get_model_path_key()` (which was making the tests fail) is being used only in the testsuite and never in the actual production code. As such, shouldnt be its definition moved into the testusite as well? Or maybe replaced with usage of `get_model_key()`.

Yeah. I didn't notice this before. I think we can move it to the tests part.

File: tests/common.py
Comment (kemingy) [inline tests/common.py:142]:
```suggestion
    generate short hashed suffix for module path (instead of its path to avoid
    code-structure leaking)
```

File: Makefile
Comment (kemingy) [inline Makefile:9]:
```suggestion
	pip install falcon --upgrade
```
The reason I put "falcon" here is that falcon2 is incompatible with falcon3. So I need to run tests for both of them.

I think starlette doesn't have this issue?

Comment (yedpodtrzitko) [reply Makefile:9]: yes I assumed it's to test multiple Falcon's versions. The problem I stumbled upon was that the tests require Starlette to be installed, otherwise it fails with `ImportError`, so I hijacked this to install starlette as needed.

Comment (kemingy) [reply Makefile:9]: I guess you mean when you run the `make test` in your machine, the "starlette" is required?

Actually, you can run `make install`. So all the required libraries will be installed.

CI:
* make install
* make lint
* make test

Comment (yedpodtrzitko) [reply Makefile:9]: > Actually, you can run make install

that's fair, I'll revert this then

**Review by kemingy — APPROVED** [review]

**Review by kemingy — APPROVED** [review]

**Comment by yedpodtrzitko** [comment]
wait a second with the potential merge please

**Comment by yedpodtrzitko** [comment]
ok it's good to go now on my end

**Review by kemingy — APPROVED** [review]

**Pull request merged**
